tools/WebhookProcessor/github_webhook_processor.php

```diff
@@ -217,8 +217,8 @@ function tag_pr($payload, $opened) {
 			$tags[] = $tag;
 
 	check_tag_and_replace($payload, '[dnm]', 'Do Not Merge', $tags);
-	if(!check_tag_and_replace($payload, '[wip]', 'Work In Progress', $tags))
-		check_tag_and_replace($payload, '[ready]', 'Work In Progress', $remove);
+	if(!check_tag_and_replace($payload, '[wip]', 'Work In Progress', $tags) && check_tag_and_replace($payload, '[ready]', 'Work In Progress', $remove))
+		$tags[] = 'Needs Review';
```

Comment:
This shouldn't be able to bypass PRs with outstanding changes requested reviews

Reply:
The way I understand it, this is run before the check_ready_for_review function, so the label will be removed with outstanding changes requested reviews.

https://github.com/tgstation/tgstation/blob/317a93cc1bd2f3d59fdbd183cc3c0653bb1128d2/tools/WebhookProcessor/github_webhook_processor.php#L377-L379

Reply:
Labels are still applied at the end of this function though, so you'll end up with events like `tgstation-server added 'Needs Review' and removed 'Needs Review' 10 minutes ago`

Reply:
So then the solution is to remove the apisend from both functions, and instead have them return the labels they want to add, and the labels they want to remove, then add this block to the end of the synchronize case. (Or put it in its own function)
https://github.com/JJRcop/tgstation/blob/317a93cc1bd2f3d59fdbd183cc3c0653bb1128d2/tools/WebhookProcessor/github_webhook_processor.php#L225-L236

im a little busy so i'll do this later

Reply:
https://github.com/tgstation/tgstation/pull/32062/commits/3f9e1f6cf108bcbab05ed60c1207fb4cd254fba1

Reply:
This issue was addressed.